app/controllers/api/v2/hosts_controller.rb

```diff
@@ -123,7 +123,25 @@ def destroy
       eos
 
       def status
-        render :json => { :status => @host.host_status }.to_json if @host
+        Foreman::Deprecation.api_deprecation_warning('The /status route is deprecated, please use the new /status/configuration instead')
+        render :json => { :status => @host.get_status(HostStatus::ConfigurationStatus).to_label }.to_json if @host
+      end
+
+      api :GET, "/hosts/:id/status/:type", N_("Get status of host")
+      param :id, :identifier_dottable, :required => true
+      param :type, [ HostStatus::Global ] + HostStatus.status_registry.to_a.map { |s| s.humanized_name }, :required => true, :desc => N_('
+status type, can be one of configuration
+* global
+* configuration
+* build')
+      description N_('Returns string representing a host status of a given type')
+      def get_status
```

Comment:
https://github.com/theforeman/foreman/pull/2580/files#diff-a7b88f691f88ff3665fe4f0f563fb4d6R183 - or do you want me to cover more status types?